### DISCUSSION TITLE
jwt_encode is not working

### DISCUSSION THREAD

**Discussion by yingshaoxo** [golang-jwt/jwt#282] [Q&A] [answered]
The `token_string` returns nothing

___

```go
func Jwt_encode(data map[string]interface{}, secret_string string) string {
	new_data := jwt.MapClaims{}
	for key, value := range data {
		new_data[key] = value
	}

	token := jwt.NewWithClaims(jwt.SigningMethodHS256, new_data)
	token_string, _ := token.SignedString(secret_string)

	return token_string
}
```

```go
package main

import (
	"log"
	"testing"

	"github.com/yingshaoxo/gopython/jwt_tool"
)

func Test_jwt(t *testing.T) {
	secret := "no way"

	data := make(map[string]interface{})
	data["user"] = "yingshaoxo"

	jwt_string := jwt_tool.Jwt_encode(data, secret)
	log.Println(jwt_string)

	new_data, _ := jwt_tool.Jwt_decode(jwt_string, secret)

	log.Println(data)
	log.Println(new_data)
}
```

**Comment by oxisto** [comment, marked as answer]
You are using `string` as the key for an HMAC. The signing method expects a `[]byte` as seen in the README "The [HMAC signing method](https://pkg.go.dev/github.com/golang-jwt/jwt/v5#SigningMethodHMAC) (HS256,HS384,HS512) expect []byte values for signing and validation" or in the example:
https://github.com/golang-jwt/jwt/blob/0d2f0d4809b8a7e01b0cb27cbfde79a69cb2a298/example_test.go#L16-L29

Note, that you are intentionally ignoring the error of `SignedString`. This error contains more information why it is failing and this case would you provide with an error that you are using the wrong key type for this signing algorithm.

**Comment by yingshaoxo** [comment]
Alright, thanks, now I've solved this problem:

```golang
func Jwt_encode(data map[string]interface{}, secret_string string) string {
	new_data := jwt.MapClaims{}
	for key, value := range data {
		new_data[key] = value
	}

	secret_string_bytes := string_tool.String_to_bytes(secret_string)

	token := jwt.NewWithClaims(jwt.SigningMethodHS256, new_data)
	token_string, _ := token.SignedString(secret_string_bytes)

	return token_string
}
```
